### ISSUE TITLE
Can't figure out why Performance_RNN won't train

### ISSUE THREAD

**Issue by chrislos** [tensorflow/magenta#1508]
Hey everyone,
I'm a bit confused that I'm not able to train a performance_rnn with config=performance on my paperspace. My machine is running on Ubuntu 18.04.1 LTS with a cuda supported quattro q5000 card.

The only steps I did so far is 
1. installing anaconda.
2 create an python 3.6.8 env
3. pip install tensorflow-gpu
4. pip install magenta-gpu

I've been able to train a polyphonic model on that configuration.
Somewhore a perfromance_rnn stops to train with the follwing console lines:

May you have a hint what I did wrong?

thanks in advance,
christian

`
 performance_rnn_train \
> --config=performance \
> --run_dir=/home/paperspace/Magenta/yamaha_bach/tmp/train \
> --sequence_example_file=/home/paperspace/Magenta/yamaha_bach/tmp/notesequences.tfrecord \
> --hparams="rnn_layer_sizes=[512,512,512]" \
> --num_training_steps=300
/home/paperspace/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

INFO:tensorflow:hparams = {'batch_size': 64, 'rnn_layer_sizes': [512, 512, 512], 'dropout_keep_prob': 1.0, 'attn_length': 0, 'clip_norm': 3, 'learning_rate': 0.001, 'residual_connections': False, 'use_cudnn': False}
INFO:tensorflow:Train dir: /home/paperspace/Magenta/yamaha_bach/tmp/train/train
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/common/sequence_example_lib.py:94: string_input_producer (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensor_slices(string_tensor).shuffle(tf.shape(input_tensor, out_type=tf.int64)[0]).repeat(num_epochs)`. If `shuffle=False`, omit the `.shuffle(...)`.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/input.py:278: input_producer (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensor_slices(input_tensor).shuffle(tf.shape(input_tensor, out_type=tf.int64)[0]).repeat(num_epochs)`. If `shuffle=False`, omit the `.shuffle(...)`.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/input.py:190: limit_epochs (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.from_tensors(tensor).repeat(num_epochs)`.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/input.py:199: QueueRunner.__init__ (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/input.py:199: add_queue_runner (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/input.py:202: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/common/sequence_example_lib.py:95: TFRecordReader.__init__ (from tensorflow.python.ops.io_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.TFRecordDataset`.
INFO:tensorflow:Counting records in /home/paperspace/Magenta/yamaha_bach/tmp/notesequences.tfrecord.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/common/sequence_example_lib.py:150: tf_record_iterator (from tensorflow.python.lib.io.tf_record) is deprecated and will be removed in a future version.
Instructions for updating:
Use eager execution and: 
`tf.data.TFRecordDataset(path)`
INFO:tensorflow:Number of records is at least 100.
INFO:tensorflow:[<tf.Tensor 'random_shuffle_queue_Dequeue:0' shape=(?, 356) dtype=float32>, <tf.Tensor 'random_shuffle_queue_Dequeue:1' shape=(?,) dtype=int64>, <tf.Tensor 'random_shuffle_queue_Dequeue:2' shape=() dtype=int32>]
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/common/sequence_example_lib.py:134: batch (from tensorflow.python.training.input) is deprecated and will be removed in a future version.
Instructions for updating:
Queue-based input pipelines have been replaced by `tf.data`. Use `tf.data.Dataset.batch(batch_size)` (or `padded_batch(...)` if `dynamic_pad=True`).
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/models/shared/events_rnn_graph.py:52: BasicLSTMCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/models/shared/events_rnn_graph.py:65: MultiRNNCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is equivalent as tf.keras.layers.StackedRNNCells, and will be replaced by that in Tensorflow 2.0.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/models/shared/events_rnn_graph.py:272: dynamic_rnn (from tensorflow.python.ops.rnn) is deprecated and will be removed in a future version.
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/rnn.py:626: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/tensor_array_ops.py:162: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/magenta/models/shared/events_rnn_graph.py:319: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, use
    tf.py_function, which takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means `tf.py_function`s can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
    
INFO:tensorflow:Starting training loop...
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
2019-04-15 06:03:40.499610: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-04-15 06:03:40.617702: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-04-15 06:03:40.618195: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x563bf4a31220 executing computations on platform CUDA. Devices:
2019-04-15 06:03:40.618218: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): Quadro P5000, Compute Capability 6.1
2019-04-15 06:03:40.620472: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2600010000 Hz
2019-04-15 06:03:40.621089: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x563bf4a305f0 executing computations on platform Host. Devices:
2019-04-15 06:03:40.621114: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-04-15 06:03:40.621465: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: Quadro P5000 major: 6 minor: 1 memoryClockRate(GHz): 1.7335
pciBusID: 0000:00:05.0
totalMemory: 15.90GiB freeMemory: 15.48GiB
2019-04-15 06:03:40.621491: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-04-15 06:03:40.622057: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-04-15 06:03:40.622078: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2019-04-15 06:03:40.622085: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2019-04-15 06:03:40.622392: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 15056 MB memory) -> physical GPU (device: 0, name: Quadro P5000, pci bus id: 0000:00:05.0, compute capability: 6.1)
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
WARNING:tensorflow:From /home/paperspace/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py:809: start_queue_runners (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
Segmentation fault`

**Comment by cghawthorne** [comment]
A segmentation fault likely indicates an issue with your TensorFlow installation or maybe your GPU drivers. Can you try uninstalling the magenta-gpu and tensorflow-gpu packages and installing the CPU-only versions? If those work, then we know for sure it's a bug with TensorFlow and/or the GPU drivers.

**Comment by chrislos** [comment]
Hey Curtis,
sorry for my late reply. 
Thanks for your response.
I'm currently a bit busy working abroad. I'll check it out as soon as I find some time!

Best
C.
